src/Microsoft.Extensions.ApplicationModelDetection/AppModelDetector.cs

```diff
@@ -0,0 +1,220 @@
+{
+    public class AppModelDetector
+    {
+        // We use Hosting package to detect AspNetCore version
+        // it contains light-up implemenation that we care about and
+        // would have to be used by aspnet core web apps
+        private const string AspNetCoreAssembly = "Microsoft.AspNetCore.Hosting";
+
+        public AppModelDetectionResult Detect(DirectoryInfo directory)
+        {
+            var result = new AppModelDetectionResult();
+            string entryPoint = null;
+
+            // Try reading web.config and resolving framework and app path
+            var webConfig = new FileInfo(Path.Combine(directory.FullName, "web.config"));
+            if (webConfig.Exists)
+            {
+                if (TryParseWebConfig(webConfig, out var framework, out entryPoint))
+                {
+                    result.Framework = framework;
+                }
+                else
+                {
+                    // web.config exists so default to full framework
+                    result.Framework = RuntimeFramework.DotNetFramework;
+                }
+            }
+
+            // If we found entry point let's look for .deps.json
+            // in some cases it exists in desktop too
+            FileInfo depsJson = null;
+            FileInfo runtimeConfig = null;
+
+            if (!string.IsNullOrWhiteSpace(entryPoint))
+            {
+                depsJson = new FileInfo(Path.ChangeExtension(entryPoint, ".deps.json"));
+                runtimeConfig = new FileInfo(Path.ChangeExtension(entryPoint, ".runtimeconfig.json"));
+            }
+
+            if (depsJson == null || !depsJson.Exists)
+            {
+                depsJson = directory.GetFiles("*.deps.json").FirstOrDefault();
+            }
+
+            if (runtimeConfig == null || !runtimeConfig.Exists)
+            {
+
+                runtimeConfig = directory.GetFiles("*.runtimeconfig.json").FirstOrDefault();
+            }
+
+            if (depsJson != null &&
+                depsJson.Exists  &&
+                TryParseDependencies(depsJson, out var aspNetCoreVersion))
+            {
+                result.AspNetCoreVersion = aspNetCoreVersion;
+            }
+
+            // Lets try assembly version
+            if (string.IsNullOrWhiteSpace(result.AspNetCoreVersion))
+            {
+                var aspNetCoreDll = new FileInfo(Path.Combine(directory.FullName, AspNetCoreAssembly + ".dll"));
+                if (aspNetCoreDll.Exists &&
+                    TryParseAssembly(aspNetCoreDll, out aspNetCoreVersion))
+                {
+                    result.AspNetCoreVersion = aspNetCoreVersion;
+                }
+            }
+
+            if (result.Framework == RuntimeFramework.DotNetCore &&
+                runtimeConfig != null &&
+                runtimeConfig.Exists &&
+                TryParseRuntimeConfig(runtimeConfig, out var runtimeVersion))
+            {
+                result.FrameworkVersion = runtimeVersion;
```

Comment:
You never set FrameworkVersion for .NET Framework. You can get it from the apps deps.json

Reply:
Not sure if anyone needs it, and deps wouldn't always be available for  .NET Framework